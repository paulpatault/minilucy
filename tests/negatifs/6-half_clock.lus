(* Erreur de clock : exemple du cours *)

node check (x: int) returns (o: bool);
var half : bool; n : int;
let
  half = (true) -> (not (pre half));
  n = x + (x when half);
  o = true;
tel
