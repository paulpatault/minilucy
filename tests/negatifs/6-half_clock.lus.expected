
/**************************************/
/* Sugar automaton print              */
/**************************************/
type inductive_bool =
  | False
  | True

node check(x : int) returns (o : bool);
var half : bool; n : int;
let
  (half) = (true) -> (~(pre (half)));
  (n) = add(x, x when True(half));
  (o) = true;
tel

[31m[1mTyping error: [0munbound node main0

